### PR TITLE
More Vert.x ITs

### PR DESCRIPTION
**Pull request by cescoffier**
Add ITs for the DNS client
Add ITs for the TCP server and client
Add ITs for the TCP server and client using TLS

### CODE REVIEW

File: integration-tests/vertx/src/main/java/io/quarkus/it/vertx/tcp/VertxTcpEndpoint.java [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+                });
+    }
+
+    @POST
+    @Path("/ssl")
+    public CompletionStage<String> ssl(String body) {
+        System.out.println("Got request on /ssl");
+        return ssl.connect(4322, "localhost")
+                .thenCompose(socket -> {
+                    System.out.println("got ssl socket");
```
Comment (gastaldi) [inline integration-tests/vertx/src/main/java/io/quarkus/it/vertx/tcp/VertxTcpEndpoint.java]: Remove the System.out.printlns

Comment (cescoffier) [reply integration-tests/vertx/src/main/java/io/quarkus/it/vertx/tcp/VertxTcpEndpoint.java]: Sorry... that was while debugging the native thingy.

**Review by gastaldi — APPROVED** [review]

**Comment by geoand** [comment]
@cescoffier I think you'll also need to increase the timeout for the native job since it got cancelled

**Comment by cescoffier** [comment]
it has been done in https://github.com/quarkusio/quarkus/pull/6700. I will rebase. 
BTW, the few tests I've added should not impact too much the build time (no new dependencies) or the runtime (run in less than a second on my very broken machine).

**Comment by geoand** [comment]
CI always complains about even the littlest thing :P

**Comment by gsmet** [comment]
Rebased to trigger CI.

**Comment by gsmet** [comment]
Rebased to trigger a new CI run.

**Comment by gsmet** [comment]
I wonder if there's not something fishy with that one being cancelled after 40 mn.

**Comment by gsmet** [comment]
I force-pushed a rebase, let's wait for CI.

**Comment by paulrobinson** [comment]
@cescoffier, @gsmet any chance we can get this merged soon? I'm trying to get the last set of smoke tests completed.

**Comment by gastaldi** [comment]
Test failure in Windows JDK 11 :

```
[INFO] Running io.quarkus.it.vertx.TcpTest
2020-03-31 22:16:57,542 WARN  [io.net.cha.DefaultChannelPipeline] (vert.x-eventloop-thread-0) An exceptionCaught() event was fired, and it reached at the tail of the pipeline. It usually means the last handler in the pipeline did not handle the exception.: java.lang.NullPointerException
	at io.vertx.core.net.impl.NetClientImpl.connected(NetClientImpl.java:216)
	at io.vertx.core.net.impl.NetClientImpl.lambda$doConnect$3(NetClientImpl.java:187)
	at io.vertx.core.net.impl.ChannelProvider.lambda$connect$1(ChannelProvider.java:78)
	at io.vertx.core.net.impl.ChannelProvider$1.userEventTriggered(ChannelProvider.java:113)
	at io.netty.channel.AbstractChannelHandlerContext.invokeUserEventTriggered(AbstractChannelHandlerContext.java:344)
	at io.netty.channel.AbstractChannelHandlerContext.invokeUserEventTriggered(AbstractChannelHandlerContext.java:330)
	at io.netty.channel.AbstractChannelHandlerContext.fireUserEventTriggered(AbstractChannelHandlerContext.java:322)
	at io.netty.handler.ssl.SslHandler.setHandshakeSuccess(SslHandler.java:1745)
	at io.netty.handler.ssl.SslHandler.unwrap(SslHandler.java:1404)
	at io.netty.handler.ssl.SslHandler.decodeJdkCompatible(SslHandler.java:1219)
	at io.netty.handler.ssl.SslHandler.decode(SslHandler.java:1266)
	at io.netty.handler.codec.ByteToMessageDecoder.decodeRemovalReentryProtection(ByteToMessageDecoder.java:498)
	at io.netty.handler.codec.ByteToMessageDecoder.callDecode(ByteToMessageDecoder.java:437)
	at io.netty.handler.codec.ByteToMessageDecoder.channelRead(ByteToMessageDecoder.java:276)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:377)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:363)
	at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:355)
	at io.netty.channel.DefaultChannelPipeline$HeadContext.channelRead(DefaultChannelPipeline.java:1410)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:377)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:363)
	at io.netty.channel.DefaultChannelPipeline.fireChannelRead(DefaultChannelPipeline.java:919)
	at io.netty.channel.nio.AbstractNioByteChannel$NioByteUnsafe.read(AbstractNioByteChannel.java:163)
	at io.netty.channel.nio.NioEventLoop.processSelectedKey(NioEventLoop.java:714)
	at io.netty.channel.nio.NioEventLoop.processSelectedKeysOptimized(NioEventLoop.java:650)
	at io.netty.channel.nio.NioEventLoop.processSelectedKeys(NioEventLoop.java:576)
	at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:493)
	at io.netty.util.concurrent.SingleThreadEventExecutor$4.run(SingleThreadEventExecutor.java:989)
	at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74)
	at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
	at java.base/java.lang.Thread.run(Thread.java:834)

[ERROR] Tests run: 2, Failures: 0, Errors: 1, Skipped: 0, Time elapsed: 1,800.12 s <<< FAILURE! - in io.quarkus.it.vertx.TcpTest

**Comment by cescoffier** [comment]
So.. unfortunately, things are broken now:

1. Vert.x DNS cannot be used anymore in native. This is a change in Netty I guess.
2. NetClient fails (we should be able to fix this one). 

About 1, several weeks / months have passed, it will be hard to find the reason.

**Comment by gsmet** [comment]
So should we close this PR? Or do we need to pursue this somehow?

**Comment by cescoffier** [comment]
Unfortunately yes, I wanted to revisit it, never had the time.

**Pull request closed without merging**